### PR TITLE
Refactor SQS Client

### PR DESCRIPTION
**Pull request by MancunianSam**
The usual changes. Turn the class into a trait with an implementation. This makes it easier to test.

I've also changed the send message method to allow us to add a message group id and a deduplication id.

### CODE REVIEW

File: sqs/src/test/scala/uk/gov/nationalarchives/DASQSClientTest.scala [hunk 1]
```diff
@@ -36,20 +39,23 @@ class DASQSClientTest extends AnyFlatSpec with MockitoSugar {
     val mockResponse = CompletableFuture.completedFuture(SendMessageResponse.builder().build())
     when(sqsAsyncClient.sendMessage(sendMessageCaptor.capture())).thenReturn(mockResponse)
 
-    val client = new DASQSClient[IO](sqsAsyncClient)
-    client.sendMessage("https://test")(Test("testMessage", "testValue")).unsafeRunSync()
+    val fifoConfig = FifoQueueConfiguration("MessageGroupId", "MessageDeduplicationId")
+    val client = DASQSClient[IO](sqsAsyncClient)
+    client.sendMessage("https://test")(Test("testMessage", "testValue"), Option(fifoConfig)).unsafeRunSync()
```
Comment (techncl) [inline sqs/src/test/scala/uk/gov/nationalarchives/DASQSClientTest.scala:44]: Should we add a test or add table (and do a `forAll`) for the case where a `fifoConfig` was not passed in?

File: sqs/src/main/scala/uk/gov/nationalarchives/DASQSClient.scala [hunk 2]
```diff
@@ -90,18 +72,65 @@ class DASQSClient[F[_]: Async](sqsAsyncClient: SqsAsyncClient):
...
+      val receiveRequest = ReceiveMessageRequest.builder
+        .queueUrl(queueUrl)
+        .maxNumberOfMessages(maxNumberOfMessages)
+        .build
+
+      Async[F]
+        .fromCompletableFuture(Async[F].pure(sqsAsyncClient.receiveMessage(receiveRequest)))
+        .flatMap { response =>
+          response.messages.asScala.toList
+            .map(message =>
```
Comment (techncl) [inline sqs/src/main/scala/uk/gov/nationalarchives/DASQSClient.scala:119]: This could be a `.traverse`

**Review by techncl — COMMENTED** [review]
Looks good! Just a few comments; also, should we update the SQS client documentation?

File: sqs/src/test/scala/uk/gov/nationalarchives/DASQSClientTest.scala (hunk 1 above)
Comment (MancunianSam) [reply sqs/src/test/scala/uk/gov/nationalarchives/DASQSClientTest.scala:44]: Good point. I've added another test in, using forAll seems a bit much.

File: sqs/src/main/scala/uk/gov/nationalarchives/DASQSClient.scala (hunk 2 above)
Comment (MancunianSam) [reply sqs/src/main/scala/uk/gov/nationalarchives/DASQSClient.scala]: Good spot, cheers. I've changed it.

**Review by techncl — DISMISSED** [review]
👍

**Review by techncl — APPROVED** [review]
👍

**Pull request merged**